app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java

```diff
@@ -197,6 +205,33 @@ private static void restoreDefaultSettings(HashMap<String, Object> settings) {
         FileUtil.writeFile(SETTINGS_FILE.getAbsolutePath(), new Gson().toJson(settings));
     }
 
+    public static boolean getRootAccess() {
+        Process p;
+        try {
+            // Preform su to get root privledges
+            p = Runtime.getRuntime().exec("su");
+
+            // Attempt to write a file to a root-only
+            DataOutputStream os = new DataOutputStream(p.getOutputStream());
+            os.writeBytes("echo \"Do I have root?\" >/system/sd/temporary.txt\n");
```

Comment:
Did you test this out? Writing to a process' output stream seems suspicious. Also, what if `/system` was mounted read-only? That'd false-positively assume that root isn't available. Just checking if executing `su` works is enough imo.

Reply:
Maybe a library for checking for root access is better. I'm pretty sure there are some, maybe even by Magisk's developer. I'll check.

Reply:
This solution was found on StackOverflow, however I have tested it on 3 smartphones and it works

Reply:
It is a good feature